app/bindings/api/v0/swagger.json

```diff
@@ -325,18 +341,46 @@
           "type": "integer",
           "description": "The event's numerical order within this session. E.g. the first event after a session is started should give 0 for this field, the next one should give 1, etc."
         },
-        "page_id": {
+        "target_id": {
           "type": "string",
-          "description": "The page id."
+          "description": "The target element id."
         },
-        "book_id": {
+        "target_type": {
           "type": "string",
-          "description": "The book id."
+          "description": "The target element (interactive) type."
+        },
+        "target_attributes": {
+          "type": "string",
```

Comment:
in your avro schemas you have these `attributes` objects as `map` types, but this is a string, is this wrong? or is there a conversion happening somewhere? can it be an object in the swagger so js can just pass the object through?

Reply:
Looks like I went with highlights/source_metadata, rather than the way I defined source_uri/query. Since we can't hard code the keys, we can't use a record type (like source_uri itself) but I think I can make it a map w/ array of string, like the source_uri/query part.

Reply:
Hmm, do we need to fix highlights/source_metadata, as well?

Reply:
Changed to array(:string) so the avro will parse it correctly. Looks like the swagger side still is String for all these (including source_uri) so I guess we're already auto-converting these.

Reply:
i see that the source_metadata is coming through correctly, but the way that you're setting up the attributes doesn't seem to be the same as the source_metadata, so i'm confused 

<img width="858" alt="image" src="https://user-images.githubusercontent.com/636227/201697798-0210ccb4-61af-4048-a613-b3002831e50b.png">

Reply:
Right, now I'm setting it up like source_uri/query, how how does that look?

Reply:
Gah, need to turn notifications on for github, I keep missing your replies, so this conversation is dragging. I jumped on the avro map part of this, and found a problem, completely ignored the actual line of swagger.rb you were commenting on. I agree, making it an object makes sense, I got distracted by the avro map parts. Build w/ change now, will deploy this after lunch,